### ISSUE TITLE
[Android] OutOfMemoryError

### ISSUE THREAD

**Issue by KSchau** [alexrainman/CarouselView#308]
When I open the carousel, flip through a few images from the 100+ item ItemsSource, and then change the ItemsSource via the binding, I get an OutOfMemoryError after the first few swipes through the new ItemsSource. Keeping track of memory, it looks like there are some references being held onto in the Carousel that are preventing those images being GC'd and they are hanging around.

My XAML looks like:

``` XAML
<controls:CarouselViewControl x:Name="Carousel"
    ItemsSource="{Binding SelectedCategory.ArticlePages}"
    Position="{Binding Position}"
    InterPageSpacing="10"
    HorizontalOptions="FillAndExpand">
```

And I change the `SelectedCategory.ArticlePages` object in the background (it's an `ObservableCollection<ArticlePageViewModel>` ).

The error I am getting is:

java.lang.OutOfMemoryError: Failed to allocate a 24174604 byte allocation with 4190184 free bytes and 19MB until OOM
	at dalvik.system.VMRuntime.newNonMovableArray(Native Method)
	at android.graphics.BitmapFactory.nativeDecodeStream(Native Method)
	at android.graphics.BitmapFactory.decodeStreamInternal(BitmapFactory.java:646)
	at android.graphics.BitmapFactory.decodeStream(BitmapFactory.java:622)

If you could either point me to how I can avoid this issue or address it in the control, that would be phenomenal!

EDIT: I am using FFImageLoading in the item template and have scaled down the images to what seems like a reasonable size - I have tried nulling out the ItemsSource bound object, setting it to a new ObservableCollection, and looking for a way to clear the memory but none of those have worked.

**Comment by alexrainman** [comment]
Which version of the control? This has been corrected in the latest one.

**Comment by KSchau** [comment]
Using CarouselView.FormsPlugin version 5.1.2 - the issue I had before where each image added to memory by a significant amount is gone but this one popped up now.

**Comment by alexrainman** [comment]
Just released version 5.1.3 please test.

**Comment by KSchau** [comment]
Ok, with the new version I am seeing some improvement over 5.1.2 I think, I still have 1 issue but I think I can resolve it on my end. Thank you for the prompt response again!

**Comment by alexrainman** [comment]
And whats that issue u still have? I am kind of done with this control. It will be included in Forms and i am done. Dont have time for this anymore :)

**Comment by KSchau** [comment]
When I changed the ItemsSource, I hadn't realized the images being pointed to by the new image source were significantly higher resolution and needed to downscale accordingly. I am downscaling more now and it appears to be working on my emulator at least. I'm glad they picked your implementation for Forms - it's by far the best one I tried with only two tiny issues - one of which you had fixed before I even had a chance to report it! With it being integrated into Forms, will the Xamarin team be taking over further development?

Thanks for all the quick responses, it's rare to see responses posted this quickly for Xamarin controls!
Cheers!
